service/resource/deployment/resource.go

```diff
@@ -188,18 +198,10 @@ func (r *Resource) ProcessCreateState(obj, createState interface{}) error {
 		for _, deploy := range deploymentsToCreate {
 			r.logger.Log("cluster", key.ClusterID(customObject), "debug", fmt.Sprintf("creating deployment %s", deploy.Name))
 
-			params := make(map[string]interface{}, len(deploy.Parameters))
-			for key, val := range deploy.Parameters {
-				params[key] = struct {
-					Value interface{}
-				}{
-					Value: val,
-				}
-			}
 			deployment := azureresource.Deployment{
```

Comment:
IMO the deployment is the resource state. So all this should just be injected in here and the process method transforms the interface to the proper type and executes client logic against it. I also assumed we fix this separately.

Reply:
The problem is the Azure API returns a DeploymentExtended object from the Get API request but takes in a Deployment object. So I went with implementing an internal structure in https://github.com/giantswarm/azure-operator/blob/master/service/resource/deployment/spec.go

https://godoc.org/github.com/Azure/azure-sdk-for-go/arm/resources/resources#DeploymentsClient.Get
https://godoc.org/github.com/Azure/azure-sdk-for-go/arm/resources/resources#DeploymentsClient.CreateOrUpdate

I can switch to using the Deployment object as the resource but if so I'd like to do later in a separate issue.

Reply:
IMO the state/structure used to act against some client is the one a resource should dispatch and manage. When this is your custom Deployment defined in the spec this is ok. Then this should also be the current, desired, create and delete state. Fine with fixing this separately.

Reply:
Ack but I think this is done. All the methods pass a []Deployment so we can deploy multiple templates later if needed. The logic that processed the parameters is now called from `GetDesiredState` and not in `ProcessCreateState`.

The delete state is not implemented because deleting the resource group deletes all resources including the deployments.